helix-term/src/ui/editor.rs

```diff
@@ -441,6 +441,12 @@ impl EditorView {
             // extra loops if the code is deeply nested.
 
             for i in starting_indent..(indent_level / tab_width as u16) {
+                if config.indent_guides.rainbow {
+                    let color_index: usize = i as usize % theme.rainbow_length();
+                    indent_guide_style =
+                        indent_guide_style.patch(theme.get(&format!("rainbow.{}", color_index)));
+                }
```

Comment:
The new `mut`s in this change can be eliminated

```suggestion
                let style = if config.indent_guides.rainbow {
                    let color_index = i as usize % theme.rainbow_length();
                    indent_guide_style.patch(theme.get(&format!("rainbow.{}", color_index)))
                } else {
                    indent_guide_style
                };
```

Saving that patched style in `indent_guide_style` has a problem that you'll patch each color of the rainbow over the last colors (for example, the first indent guide is `red + indent_guide_style`, then the second is `yellow + red + indent_guide_style`, etc. when we only want `rainbow_color + indent_guide_style`)